src/tests/command_handler.cxx

```diff
@@ -0,0 +1,100 @@
+#include "../CommandHandler.h"
+#include <fstream>
+#include <gtest/gtest.h>
+#include <rapidjson/filereadstream.h>
+#include <sstream>
+
+using namespace FileWriter;
+
+namespace FileWriter {
+std::string find_filename(rapidjson::Document const &);
+std::string find_job_id(rapidjson::Document const &);
+std::string find_broker(rapidjson::Document const &);
+std::vector<std::unique_ptr<Source>> find_source(rapidjson::Document const &,
+                                                 rapidjson::Document const *);
+ESSTimeStamp find_time(rapidjson::Document const &, const std::string &);
+} // namespace FileWriter
+
+class CommandHandler_Test : public ::testing::Test {
+
+protected:
+  virtual void SetUp() {
+    auto s = parse_impl("tests/msg-cmd-new-00.json");
+    new_00.Parse(s.c_str());
+    s = parse_impl("tests/msg-cmd-new-01.json");
+    new_01.Parse(s.c_str());
+    s = parse_impl("tests/msg-cmd-new-03.json");
+    new_03.Parse(s.c_str());
+  }
+
+  rapidjson::Document new_00;
+  rapidjson::Document new_01;
+  rapidjson::Document new_03;
+
+private:
+  std::string parse_impl(const std::string &fname) const {
+    std::ifstream t(fname);
+    std::stringstream buffer;
+    buffer << t.rdbuf();
+    std::string cmd{buffer.str()};
+    LOG(7, "cmd: {}", cmd.c_str());
+    return cmd;
+  }
+};
+
+TEST_F(CommandHandler_Test, filename) {
```

Comment:
Running unit tests can catch cases where the output is not what the programmer intended. However another cause of bugs is actually that the programmer was mistaken about what the output should be. An important role of unit tests is therefore to document the intention of the programmer so that other people can spot such mistakes. For this reason I think it is a really good idea to have readable test names which make it clear what is being tested. As an example, this test could be called `test_found_filename_matches_that_in_json`. Consider renaming this and the other tests.